### PR TITLE
Extend proxy support

### LINKED ISSUE #130: Allow the application to work behind a proxy
The baseUrl for the API and the UI should be configurable.
This will allow the proxy to work.
You should be able to set the baseUrl up front.
Example: `hostname.com:xxxx/sub1/sub2/`
If the baseUrl does not contain a trailing forward slash, a forward slash will be appended.

1. update config.js to not use location.orgin but take the correct part of the location 
1. test by hand (click through a bit to find broken links)

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

closes #130 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [ ] ~Sufficient test cases (reproduces the bug/tests the requested feature)~
- [x] Correct, in line with design
- [ ] ~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: app/components/inmantaApi.js [hunk 1]
```diff
@@ -63,7 +63,12 @@ inmantaApi.service('authService', ["inmantaConfig", function Nodeservice(inmanta
 
 inmantaApi.service('inmantaService', ["$http", "inmantaConfig", "$q", "$cacheFactory", "$rootScope", "alertService", function Nodeservice($http, inmantaConfig, $q, $cacheFactory, $rootScope, alertService) {
     var inmantaAPI = {};
-    var impURL = inmantaConfig.backend + "api/v1/";
+    var pathProxyPart = location.pathname.split("/dashboard")[0].substring(1);
```
Comment (Stijnkool) [inline app/components/inmantaApi.js:66]: Why do we have `.substring(1)` at the end? To drop the first char?

File: app/partials/sideController.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ resv.controller('sideController', ['$scope', '$rootScope', 'inmantaService', 'in
 	$scope.state = $stateParams;
 	inmantaService.getConsoleAvailable().then(function (lsm) {
 		$scope.lsm = true;
+		$scope.consoleUrl = location.pathname.split("/dashboard")[0] + "/console/";
+		console.log($scope.consoleUrl);
```
Comment (Stijnkool) [inline app/partials/sideController.js:10]: Maybe remove console.log?

**Review by Stijnkool — APPROVED** [review]
Just 1 question and 1 log statement to remove.

File: app/components/inmantaApi.js (hunk 1 above)
Comment (andraskvr) [reply app/components/inmantaApi.js:66]: Yes, it starts with a slash, and the `inmantaConfig.backend` ends with slash, this is just to not have 2 slashes after each other

**Pull request closed without merging**